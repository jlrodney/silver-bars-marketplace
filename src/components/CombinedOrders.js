import React from 'react';
import Combined from './Combined';
// import uuid from 'uuid';
function combiningOrders (arrayOfOrderObjects) {
  let orderPriceQuantObject = {};
  for ( let i = 0; i < arrayOfOrderObjects.orders.length; i++) {
    if (!orderPriceQuantObject[arrayOfOrderObjects.orders[i].price]) {
      orderPriceQuantObject[arrayOfOrderObjects.orders[i].price] = arrayOfOrderObjects.orders[i].quantity;
    } else {
      orderPriceQuantObject[arrayOfOrderObjects.orders[i].price] += arrayOfOrderObjects.orders[i].quantity;
    }
  }
  return Object.entries(orderPriceQuantObject);
}

const CombinedOrders = (props) => {
  return(
    <div>
      { props.orders.length >= 1 && props.orders[0].type === "BUY" ?
          combiningOrders(props).reverse().map((order) => (
            <Combined
            key={order[0]}
            quantity={order[1]}
            price={order[0]}
            />
          )) :
          combiningOrders(props).map((order) => (
            <Combined
            key={order[0]}
            quantity={order[1]}
            price={order[0]}
            />
          ))
    }
    </div>
  )
}

export default CombinedOrders;
