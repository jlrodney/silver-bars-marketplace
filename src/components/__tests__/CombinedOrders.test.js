import React from 'react';
import { shallow, mount } from 'enzyme';
import CombinedOrders from '../CombinedOrders';

const buyOrders = [
  {
    orderId: 'xyz986',
    userId: '1',
    quantity: 22,
    price: 230,
    type: 'BUY'
  },
  {
    orderId: 'xyz988',
    userId: '1',
    quantity: 222,
    price: 230,
    type: 'BUY'
  },
  {
    orderId: 'xyz987',
    userId: '3',
    quantity: 32,
    price: 2300,
    type: 'BUY'
  }
]

const sellOrders = [
  {
    orderId: 'xyz986',
    userId: '1',
    quantity: 22,
    price: 230,
    type: 'SELL'
  },
  {
    orderId: 'xyz988',
    userId: '1',
    quantity: 25,
    price: 230,
    type: 'SELL'
  },
  {
    orderId: 'xyz987',
    userId: '3',
    quantity: 32,
    price: 2300,
    type: 'SELL'
  }
]
describe('Combined orders', () => {

  it('should have higher price component first for the two buy orders', () => {
    const wrapper = shallow(<CombinedOrders orders={buyOrders}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have lower price component first for the two sell orders', () => {
    const wrapper = shallow(<CombinedOrders orders={sellOrders}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
