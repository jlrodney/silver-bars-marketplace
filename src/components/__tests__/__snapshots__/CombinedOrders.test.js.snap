// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combined orders should have higher price component first for the two buy orders 1`] = `
<div>
  <Combined
    key="2300"
    price="2300"
    quantity={32}
  />
  <Combined
    key="230"
    price="230"
    quantity={244}
  />
</div>
`;

exports[`Combined orders should have lower price component first for the two sell orders 1`] = `
<div>
  <Combined
    key="230"
    price="230"
    quantity={47}
  />
  <Combined
    key="2300"
    price="2300"
    quantity={32}
  />
</div>
`;
